src/fabric/twist.rs

```diff
@@ -19,33 +19,36 @@ impl Fabric {
     pub fn single_twist(&mut self, spin: Spin, pretenst_factor: f32, scale_factor: f32, face_id: Option<UniqueId>) -> [(FaceName, UniqueId); 2] {
         let face = face_id.map(|id| self.face(id));
         let scale = face.map(|Face { scale, .. }| *scale).unwrap_or(1.0) * scale_factor;
-        let base = self.base_triangle(face);
-        let matrix = Self::translation_matrix_for_base(base);
-
-        let Brick { joints, intervals, faces } = Brick::new(match spin {
+        let brick = Brick::new(match spin {
             Left => BrickName::LeftTwist,
             Right => BrickName::RightTwist,
         });
-        for point in joints {
-            self.create_joint(matrix.transform_point(point));
-        }
-        for (alpha_index, omega_index, role) in intervals {
+        let matrix = face.map(|face| face.space(self));
+        let joints: Vec<usize> = brick.joints
+            .iter()
+            .map(|point| self.create_joint(match matrix {
+                None => *point,
+                Some(matrix) => matrix.transform_point(*point),
```

Comment:
```suggestion
            .into_iter()
            .map(|point| self.create_joint(match matrix {
                None => point,
                Some(matrix) => matrix.transform_point(point),
```
Can move out of `brick` no problem